matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/crypto/algorithms/megolm/MXMegolmDecryption.kt

```diff
@@ -319,11 +323,20 @@ internal class MXMegolmDecryption(private val credentials: Credentials,
                                             Try.just(Unit)
                                         }
                                         Timber.v("""## shareKeysWithDevice() : sharing keys for session ${body?.senderKey}|${body?.sessionId} with device $userId:$deviceId""")
-                                        val inboundGroupSession = olmDevice.getInboundGroupSession(body?.sessionId, body?.senderKey, body?.roomId)
 
                                         val payloadJson = HashMap<String, Any>()
                                         payloadJson["type"] = EventType.FORWARDED_ROOM_KEY
-                                        payloadJson["content"] = inboundGroupSession!!.exportKeys()!!
+
+                                        olmDevice.getInboundGroupSession(body?.sessionId, body?.senderKey, body?.roomId)
+                                                .fold(
+                                                        {
+                                                            // TODO
```

Comment:
TODO ?

Reply:
it will send event without useful content...
There is lot of ? in this code, we should rework it...